tekdoc/main.go

```diff
@@ -17,15 +17,72 @@ limitations under the License.
 package main
 
 import (
+	"flag"
 	"fmt"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1alpha1"
+	"github.com/tektoncd/pipeline/pkg/client/clientset/versioned/scheme"
+	"io/ioutil"
+	"log"
+	"os"
+	"text/template"
+)
 
-	_ "github.com/tektoncd/pipeline/pkg/pod"
+var (
+	filename = flag.String("f", "", "Name of the file to parse")
 )
+func check(err error) {
+	if err != nil {
+		panic(err)
+	}
+}
+type taskStruct struct {
+	Name   string
+	Description string
+	Default *v1alpha1.ArrayOrString
+	Type v1alpha1.ParamType
+	Type1 v1alpha1.PipelineResourceType
+}
 
 func main() {
-	fmt.Println("Hello this is STILL TekDoc")
-
 	// TODO: take a file path, parse it as YAML -> v1alpha1.Task
+	flag.Parse()
+	dat, err := ioutil.ReadFile(*filename)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	var task v1alpha1.Task
+	if _, _, err := scheme.Codecs.UniversalDeserializer().Decode(dat, nil, &task); err != nil {
+		log.Fatal(err)
+	}
+
 	// TODO: use Go templates to print v1alpha1.Task as Markdown
-	// TODO: tests!
+	fmt.Println("#",task.Name)
+	fmt.Printf("## Install the Task \n kubectl apply -f https://raw.githubusercontent.com/tektoncd/catalog/master/%s/%s.yaml\n ### Inputs \n ***Parameters :-***\n",task.Name,task.Name)
+
+	for _, i := range task.Spec.Inputs.Params {
+		InputParams := taskStruct{Name: i.Name, Description: i.Description, Default: i.Default, Type: i.Type}
+		tmpl := template.Must(template.New("Details").Parse("- Name : {{.Name}}, {{.Description}}, ({{.Default}})\n" ))
+		check(tmpl.Execute(os.Stdout, InputParams))
+	}
+
+	if task.Spec.Inputs.Resources != nil{
+		print(" ***Resources :-***\n")
+		for _, i := range task.Spec.Inputs.Resources {
+			InputParams := taskStruct{Name: i.Name, Type1: i.Type}
+			tmpl := template.Must(template.New("Details").Parse("- Name : {{.Name}} \n- Type : {{.Type1}}\n"))
+			check(tmpl.Execute(os.Stdout, InputParams))
+			}
```

Comment:
Make sure to use gofmt! https://blog.golang.org/go-fmt-your-code

There should be a plugin for whatever editor you use. (if you have trouble finding one just let us know and we can help find one)

Reply:
Added the plugin, thanks!